overlord/devicestate/devicemgr.go

```diff
@@ -169,10 +170,52 @@ func (m *DeviceManager) ensureOperational() error {
 	return nil
 }
 
+// ensureSnaps makes sure that the snaps from seed.yaml get installed
+// with the matching assertions
+func (m *DeviceManager) ensureSeedYaml() error {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// FIXME: enable on classic?
+	//
+	// Disable seed.yaml on classic for now. In the long run we want
+	// classic to have a seed parsing as well so that we can install
+	// snaps in a classic environment (LP: #1609903). However right
+	// now it is under heavy development so until the dust
+	// settles we disable it.
+	if release.OnClassic {
+		return nil
+	}
+
+	all, err := snapstate.All(m.state)
```

Comment:
Added a tech debt microtask item to our board:
* Add snapstate.IsEmpty to avoid len(snapstate.All(st)) == 0 checks